### DISCUSSION TITLE
Imagemagick chokes on larger files

### DISCUSSION THREAD

**Discussion by ChiKwasi** [ImageMagick/ImageMagick#6958] [Digital Image Processing]
Is there anything I can alter to allow imagemagick to do a task like this? Any settings anywhere I can change? This is probably an extreme in my use and not a common occurence.

User just uploaded 9 files.
Large... 2mb up to 6mb in size.

What I do is I first put them into an /originals directory, then I do some processing on them.

I reduce them with imagemagick into the final website height size, place them into a /pending_photos directory, and make thumbnails of them as well.

Seems things choked. because of the large file sizes
Only 2 files successfully were processed via Imagemagick.

I'm using dropzone.js as a drag/drop file uploader, and 6 of the files and the other 3 may have been uploaded with the group "upload" button vs one by one, but not sure on that.

Everything works fine until this situation with these large files today...

And forcing people to only submit smaller files may not be that good of an option unfortunately.

I'm using Windows Server 2019 and IIS. Plenty of Ram and CPU on the server, it's a VPS and just the one website on it.

Thoughts?

**Comment by fmw42** [comment]
Check your resources.  Imagemagick limits your resources and my restrict your RAM, width or height of images that you can process. If too low, you can change the values in policy.xml file.

`magick -list resource`

(or convert if on IM 6).

**Reply by ChiKwasi** [reply]
Resource limits:
  Width: 922.337PP
  Height: 922.337PP
  Area: 34.3061GP
  List length: unlimited
  Memory: 15.975GiB
  Map: 31.95GiB
  Disk: unlimited
  File: 1536
  Thread: 1
  Throttle: 0
  Time: unlimited
  
  The 6mb file uploaded was 4380x5475
  
  The 2 it processed and reduced etc were
  1800w x 2700h
  4380w x 5474h
  
  Would the order they get processed affect it's performance
  I'm not sure if they were uploaded 1x1 or in batches
  

 Should I increase width and height to 6000PP ??

**Comment by urban-warrior** [comment]
Try the [web-safe](https://imagemagick.org/source/policy-websafe.xml) ImageMagick security policy which includes policies that protects the system from becoming overburdened. See https://imagemagick.org/script/security-policy.php for a discussion of using security policies.

**Comment by snibgo** [comment]
> Seems things choked. because of the large file sizes

Did you get any error messages?

IM doesn't usually care about file sizes in MB. The important size is in pixels: the width and height, and area (area = width * height). What does `magick -list resource` say? If the limits are too low, raise them by editing `policy.xml`.

**Reply by ChiKwasi** [reply]
Didn't see any error messages. Where might they be?
After the images are uploaded they get resized, so I wouldn't be aware of any immediately.
